pkgs/tools/misc/rtx/default.nix

```
@@ -0,0 +1,52 @@
+{ lib
+, rustPlatform
+, fetchFromGitHub
+, installShellFiles
+, stdenvNoCC
+, perl
+, Security
+}:
+
+let
+  owner = "jdxcode";
+  pname = "rtx";
+  version = "1.22.6";
+in
+rustPlatform.buildRustPackage rec {
+  inherit pname version;
+
+  src = fetchFromGitHub {
+    inherit owner;
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "sha256-OX4GNS/T1a9qbQFTUNlfDi9r9z3iSXM2Cc1NLxASRZc=";
+  };
+
+  cargoSha256 = "sha256-+asY1zDAoiniRxCqlZOjlQMEgvjA0Y5K5SZT0y0qCRY=";
+
+  # perl because: https://docs.rs/openssl/0.10.45/openssl/#vendored
+  nativeBuildInputs = [ perl installShellFiles ];
+  buildInputs = lib.optionals stdenvNoCC.isDarwin [ Security ];
+
+  # tests initialization script fails (not any particular test)
+  # with `~/data/plugins/tiny/bin/list-aliases no such file or directory`
+  doCheck = false;
+
+  postInstall = ''
+    installManPage ./man/man1/rtx.1
+    installShellCompletion \
+      ./completions/rtx.{bash,fish} \
+      --zsh ./completions/_rtx
+  '';
+
+  passthru.updateScript = ./update.sh;
+
+  meta = rec {
+    description = "Runtime Executor (asdf rust clone)";
+    homepage = "https://github.com/jdxcode/rtx";
+    changelog = "${homepage}/releases/tag/v${version}";
+    license = lib.licenses.mit;
+    maintainers = with lib.maintainers; [ konradmalik ];
+    mainProgram = "rtx";
```

The explicit `mainProgram` is not necessary anymore since it matches the package name, and according to the review guidelines:

> Platforms should be set (or the package will not get binary substitutes). 

https://nixos.org/manual/nixpkgs/stable/#reviewing-contributions-new-packages

```suggestion
    platforms = platforms.linux ++ platforms.darwin;
```

---

mainProgram removed, platforms added